### ISSUE TITLE
Replace python-crfsuite with underthesea-core

### ISSUE THREAD

**Issue by rain1024** [undertheseanlp/underthesea#907]
## Summary

Replace `python-crfsuite` dependency with `underthesea-core` for CRF training and inference. This will reduce dependencies and use our native Rust implementation.

## Background

Benchmark results show `underthesea-core` 3.0.3 is a viable replacement:

| Task | Training | Prediction | Accuracy |
|------|----------|------------|----------|
| POS Tagging | 1.22x faster | 0.81x | identical (0.9536 vs 0.9532) |
| Word Segmentation | 1.15x faster | 1.24x faster | identical (0.9757) |

- underthesea-core can load existing .crfsuite models trained by python-crfsuite
- Accuracy is equivalent

## Files to Update

| File | Current Usage | Replace With |
|------|---------------|--------------|
| `underthesea/trainers/crf_trainer.py` | `pycrfsuite.Trainer` | `underthesea_core.CRFTrainer` |
| `underthesea/models/crf_sequence_tagger.py` | `pycrfsuite.Tagger` | `underthesea_core.CRFTagger` |
| `underthesea/models/fast_crf_sequence_tagger.py` | `pycrfsuite.Tagger` | `underthesea_core.CRFTagger` |
| `underthesea/models/py_crf_sequence_tagger.py` | `pycrfsuite.Tagger` | `underthesea_core.CRFTagger` |

## API Mapping

```python
# Training
# Before (python-crfsuite)
trainer = pycrfsuite.Trainer(verbose=True)
for xseq, yseq in zip(X_train, y_train):
    trainer.append(xseq, yseq)
trainer.set_params({"c1": 0.1, "c2": 0.01, "max_iterations": 100})
trainer.train(model_path)

# After (underthesea-core)
from underthesea_core import CRFTrainer
trainer = CRFTrainer()
trainer.set_l1_penalty(0.1)
trainer.set_l2_penalty(0.01)
trainer.set_max_iterations(100)
model = trainer.train(X_train, y_train)
model.save(model_path)

# Inference
# Before (python-crfsuite)
tagger = pycrfsuite.Tagger()
tagger.open(model_path)
tags = tagger.tag(features)

# After (underthesea-core)
from underthesea_core import CRFTagger
tagger = CRFTagger()
tagger.load(model_path)
tags = tagger.tag(features)
```

## Tasks

- [ ] Update `crf_trainer.py` to use `CRFTrainer`
- [ ] Update `crf_sequence_tagger.py` to use `CRFTagger`
- [ ] Update `fast_crf_sequence_tagger.py` to use `CRFTagger`
- [ ] Update `py_crf_sequence_tagger.py` to use `CRFTagger`
- [ ] Remove `python-crfsuite` from `pyproject.toml` dependencies
- [ ] Test with existing models (word_tokenize, pos_tag, ner, chunking)
- [ ] Update documentation if needed

## Requirements

- underthesea-core >= 3.0.3 (contains fix for CRFsuite binary format loader)
